CppSamples/Search/FindPlace/FindPlace.cpp

```diff
@@ -157,17 +158,16 @@ void FindPlace::startLocationPermission()
   QLocationPermission locationPermission{};
   locationPermission.setAccuracy(QLocationPermission::Accuracy::Precise);
   locationPermission.setAvailability(QLocationPermission::Availability::WhenInUse);
-  switch (qApp->checkPermission(locationPermission))
+
+  qApp->requestPermission(locationPermission, [this, locationPermission](const QPermission& permission)
   {
-  case Qt::PermissionStatus::Undetermined:
-    qApp->requestPermission(locationPermission, this, &FindPlace::startLocationPermission);
-    return;
-  case Qt::PermissionStatus::Granted:
-    return;
-  case Qt::PermissionStatus::Denied:
-    emit locationPermissionDenied();
-    return;
-  }
+    Q_UNUSED(permission);
+    if (qApp->checkPermission(locationPermission) == Qt::PermissionStatus::Denied)
+    {
+      emit locationPermissionDenied();
+    }
+    connectSignals();
```

Comment:
The control flow here is confusing to me. It seems like `connectSignals` is what triggers things to proceed after permissions are granted, but that doesn't seem right. Is there a way to clarify what's happening?

Reply:
Here  the `connectSignals` is called after `requestPermission` is done to avoid simultaneous execution

Reply:
I see what's happening now in connectSignals.